packages/block-library/src/navigation/edit/index.js

```diff
@@ -283,18 +283,44 @@ function Navigation( {
 			! hasResolvedNavigationMenus ||
 			isConvertingClassicMenu ||
 			fallbackNavigationMenus?.length > 0 ||
-			classicMenus?.length !== 1
+			! classicMenus?.length
 		) {
 			return;
 		}
 
 		// If there's non fallback navigation menus and
-		// only one classic menu then create a new navigation menu based on it.
-		convertClassicMenu(
-			classicMenus[ 0 ].id,
-			classicMenus[ 0 ].name,
-			'publish'
+		// a classic menu with a `primary` location or slug,
+		// then create a new navigation menu based on it.
+		// Otherwise, use the first classic menu.
+		const primaryMenus = classicMenus.filter(
+			( classicMenu ) =>
+				classicMenu.locations.includes( 'primary' ) ||
+				classicMenu.slug === 'primary'
 		);
```

Comment:
So to be clear this is

- a menu which is currently used in the location named `primary`
- a menu with the exact name of `primary`

Should we also be sure to account for capitalisation here? So convert everything to lowercase and run the test on that?

Reply:
Ah ok. If we're confident they are always lowercase then we're all good.